cpp/CMakeLists.txt

```diff
@@ -77,6 +77,7 @@ rapids_cmake_support_conda_env(conda_env MODIFY_PREFIX_PATH)
 # * compiler options ------------------------------------------------------------------------------
 rapids_find_package(
   ucx REQUIRED
+  GLOBAL_TARGETS ucx::ucp ucx::ucs ucx::uct
```

Comment:
I think such changes shouldn't be necessary, you mentioned offline they were required to build from inside another project. We'll have our CMake experts look at what would need to change here when the PR is ready for review.